src/Apps/W1/PEPPOL/App/src/Common/PEPPOL30Impl.Codeunit.al

```diff
@@ -0,0 +1,1397 @@
+            PayableAmountCurrencyID := GetSalesDocCurrencyCode(SalesHeader);
+        end;
+    end;
+
+#if not CLEAN25
+#pragma warning disable AL0432
+#endif
+    procedure GetLegalMonetaryDocAmounts(SalesHeader: Record "Sales Header"; var VATAmtLine: Record "VAT Amount Line"; var LineExtensionAmount: Text; var LegalMonetaryTotalCurrencyID: Text; var TaxExclusiveAmount: Text; var TaxExclusiveAmountCurrencyID: Text; var TaxInclusiveAmount: Text; var TaxInclusiveAmountCurrencyID: Text; var AllowanceTotalAmount: Text; var AllowanceTotalAmountCurrencyID: Text; var ChargeTotalAmount: Text; var ChargeTotalAmountCurrencyID: Text)
+#if not CLEAN25
+#pragma warning restore AL0432
+#endif
+    begin
+        LineExtensionAmount := Format(Round(VATAmtLine."VAT Base", 0.01) + Round(VATAmtLine."Invoice Discount Amount", 0.01), 0, 9);
+        LegalMonetaryTotalCurrencyID := GetSalesDocCurrencyCode(SalesHeader);
+
+        TaxExclusiveAmount := Format(Round(VATAmtLine."VAT Base", 0.01), 0, 9);
+        TaxExclusiveAmountCurrencyID := GetSalesDocCurrencyCode(SalesHeader);
+
+        TaxInclusiveAmount := Format(Round(VATAmtLine."Amount Including VAT", 0.01, '>'), 0, 9); // Should be two decimal places
+        TaxInclusiveAmountCurrencyID := GetSalesDocCurrencyCode(SalesHeader);
+
+        AllowanceTotalAmount := Format(Round(VATAmtLine."Invoice Discount Amount", 0.01), 0, 9);
+        AllowanceTotalAmountCurrencyID := GetSalesDocCurrencyCode(SalesHeader);
+        TaxInclusiveAmountCurrencyID := GetSalesDocCurrencyCode(SalesHeader);
+
+        ChargeTotalAmount := '';
+        ChargeTotalAmountCurrencyID := '';
+    end;
+
+    procedure GetLineGeneralInfo(SalesLine: Record "Sales Line"; SalesHeader: Record "Sales Header"; var InvoiceLineID: Text; var InvoiceLineNote: Text; var InvoicedQuantity: Text; var InvoiceLineExtensionAmount: Text; var LineExtensionAmountCurrencyID: Text; var InvoiceLineAccountingCost: Text)
+    begin
+        InvoiceLineID := Format(SalesLine."Line No.", 0, 9);
+        InvoiceLineNote := DelChr(Format(SalesLine.Type), '<>');
+        InvoicedQuantity := Format(SalesLine.Quantity, 0, 9);
+        InvoiceLineExtensionAmount := Format(SalesLine."VAT Base Amount" + SalesLine."Inv. Discount Amount", 0, 9);
+        LineExtensionAmountCurrencyID := GetSalesDocCurrencyCode(SalesHeader);
+        InvoiceLineAccountingCost := '';
+    end;
+
+    procedure GetLineUnitCodeInfo(SalesLine: Record "Sales Line"; var unitCode: Text; var unitCodeListID: Text)
+    var
+        UOM: Record "Unit of Measure";
+    begin
+        unitCode := '';
+        unitCodeListID := GetUNECERec20ListID();
+
+        if SalesLine.Quantity = 0 then begin
+            unitCode := UoMforPieceINUNECERec20ListIDTxt; // unitCode is required
+            exit;
+        end;
+
+        case SalesLine.Type of
+            SalesLine.Type::Item, SalesLine.Type::Resource:
```

Comment:
I would expect this code to be on the sales line. Is there any reason that it is PEPPOL specific?